### ISSUE TITLE
Set-AzureRMStorageAccount doens't receive pipeline

### ISSUE THREAD

**Issue by siebeh** [Azure/azure-powershell#8060]
When receiving the Microsoft.Azure.Commands.Management.Storage.Models.PSStorageAccount object through a pipeline the command cannot be processed.

```
PS C:\Users\x\OneDrive\Powershell\AzureTest> Get-AzureRmStorageAccount -ResourceGroupName:$RGName

StorageAccountName ResourceGroupName    Location   SkuName     Kind    AccessTier CreationTime       ProvisioningState EnableHttpsTrafficOnly
------------------ -----------------    --------   -------     ----    ---------- ------------       ----------------- ----------------------
xxxxxxxxxxxx       xxxxxxxxxxxxxxxxxxxx westeurope StandardLRS Storage            9-12-2018 16:19:27 Succeeded         False


PS C:\Users\x\OneDrive\Powershell\AzureTest> Get-AzureRmStorageAccount -ResourceGroupName:$RGName | Set-AzureRmStorageAccount -UpgradeToStorageV2
Set-AzureRmStorageAccount : ParameterBinderStrings
At line:1 char:56
+ ... urceGroupName:$RGName | Set-AzureRmStorageAccount -UpgradeToStorageV2
+                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ReadError: (:PSObject) [Set-AzureRmStorageAccount], ParameterBindingParameterDefaultValueException
    + FullyQualifiedErrorId : GetDefaultValueFailed,Microsoft.Azure.Commands.Management.Storage.SetAzureStorageAccountCommand

PS C:\Users\x\OneDrive\Powershell\AzureTest> Set-AzureRmStorageAccount -ResourceGroupName:$RGName -AccountName:$storageAccountName -UpgradeToStorageV2

StorageAccountName ResourceGroupName    Location   SkuName     Kind      AccessTier CreationTime       ProvisioningState EnableHttpsTrafficOnly
------------------ -----------------    --------   -------     ----      ---------- ------------       ----------------- ----------------------
xxxxxxxxxxxx       xxxxxxxxxxxxxxxxxxxx westeurope StandardLRS StorageV2 Hot        9-12-2018 16:19:27 Succeeded         False

PS C:\Users\x\OneDrive\Powershell\AzureTest> Get-Command Set-AzureRmStorageAccount

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Cmdlet          Set-AzureRmStorageAccount                          5.2.0      AzureRM.Storage

```

_Originally posted by @siebeh in https://github.com/_render_node/MDEyOklzc3VlQ29tbWVudDQ0NTU1MTQ5Ng==/timeline/issue_comment#issuecomment-445551496_

**Comment by cormacpayne** [comment]
This should be fixed in https://github.com/Azure/azure-powershell/pull/8068

**Comment by blueww** [comment]
Before the issue is fixed, you can workaround it by adding `-EnableHttpsTrafficOnly` parameter. Following command works with AzureRm.Storage 5.2.0 in my testing. 
`Get-AzureRmStorageAccount -ResourceGroupName $RGName | Set-AzureRmStorageAccount -EnableHttpsTrafficOnly $true -UpgradeToStorageV2`

**Comment by markcowl** [comment]
Ths fix for this is merged, it shoudl go out with Az 1.0
